runtime/autoload/health.vim

```
@@ -1,433 +1,181 @@
+"         call Bar(a:arg, "foo")
+"     endif
+" endfunction
+
+""
+" This function starts a report.
+" It should represent a general area of tests that can be understood
+" from the argument {name}
+" To start a new report, use this function again
+function! health#report_start(name) abort
+  echo '  - Checking: ' . a:name
 endfunction
 
-
-" Get version information using the specified interpreter.  The interpreter is
-" used directly in case breaking changes were introduced since the last time
-" Neovim's Python client was updated.
-function! s:version_info(python) abort
-  let pypi_version = s:latest_pypi_version()
-  let python_version = s:trim(system(
-        \ printf('"%s" -c "import sys; print(''.''.join(str(x) '
-        \ . 'for x in sys.version_info[:3]))"', a:python)))
-  if empty(python_version)
-    let python_version = 'unknown'
-  endif
-  
-  let nvim_path = s:trim(system(printf('"%s" -c "import sys, neovim;'
-        \ . 'print(neovim.__file__)" 2>/dev/null', a:python)))
-  if empty(nvim_path)
-    return [python_version, 'not found', pypi_version, 'unknown']
-  endif
-
-  let nvim_version = 'unknown'
-  let base = fnamemodify(nvim_path, ':h')
-  for meta in glob(base.'-*/METADATA', 1, 1) + glob(base.'-*/PKG-INFO', 1, 1)
-    for meta_line in readfile(meta)
-      if meta_line =~# '^Version:'
-        let nvim_version = matchstr(meta_line, '^Version: \zs\S\+')
-      endif
-    endfor
-  endfor
-
-  let version_status = 'unknown'
-  if nvim_version != 'unknown' && pypi_version != 'unknown'
-    if s:version_cmp(nvim_version, pypi_version) == -1
-      let version_status = 'outdated'
-    else
-      let version_status = 'up to date'
+""
+" Format a message for a specific report item
+function! s:format_report_message(status, msg, ...) abort
+  let l:output = '    - ' . a:status . ': ' . a:msg
+
+  " Check optional parameters
+  if a:0 > 0
+    " Suggestions go in the first optional parameter and must be a list.
+    if type(a:1) == type([])
```

If we require it to be a list, rather than silently ignoring non-list input, throw an error or else convert it to a string (`string(a:1)`).